onboard/catalog_actions.go

```diff
@@ -64,15 +88,27 @@ func (c *CatalogRepoActions) ApplyImport(ctx context.Context, it Iterator, dryRu
 			if dryRun {
 				continue
 			}
-			err := c.cataloger.CreateEntries(ctx, c.repository, DefaultBranchName, previousBatch)
-			if err != nil {
-				return nil, fmt.Errorf("failed to create batch of %d entries (%w)", len(currentBatch), err)
+			tsk := &task{
+				f: func() error {
+					return c.cataloger.CreateEntries(ctx, c.repository, DefaultBranchName, previousBatch)
+				},
+				err: new(error),
 			}
+			errs = append(errs, tsk.err)
+			wg.Add(1)
```

Comment:
This is a point of contention between workers.  Better perhaps to count just active workers have workers signal that they are done (a single call to `wg.Done()` per worker rather than per task).